src/blueapi/cli/cli.py

```diff
@@ -21,54 +19,36 @@ def main(ctx) -> None:
     if ctx.invoked_subcommand is None:
         print("Please invoke subcommand!")
 
+    config_loader = ConfigLoader(ApplicationConfig)
+
+    ctx.ensure_object(dict)
+    ctx.obj["config"] = (
+        config_loader.load_from_yaml(DEFAULT_YAML_PATH)
+        if DEFAULT_YAML_PATH.exists()
+        else config_loader.load()
+    )
+
 
 @main.command(name="worker")
-@click.option("-c", "--config", type=Path, help="Path to configuration YAML file")
-def start_worker(config: Optional[Path]):
+@click.pass_obj
+def start_worker(obj: dict) -> None:
```

Comment:
Should be able to tighten up to dict[str, BaseModel]?

Reply:
Sorry; this has since changed.
Turns out you can't modify context between child click commands. Here, 'worker' and 'controller' are child commands of 'main'. So if I pass in config to 'worker', and modify the object/context within worker, it has no difference to the context 'controller' receives. 

For now, my solution is to make 'main' take config instead. This way, config options actually propagate to both children.

In future with the REST api stuff this may not be needed, as everything the controller does now will be done via rest api. In that case we can migrate 'config' back to worker.

Reply:
Would that ever happen? The workers is the RunEngine worker with access to the devices, the Controller is the client that sends requests to the Worker remotely. Both shouldn't be called on the same invocation

Reply:
It's more that you'd expect the configuration would be shared between both.

The invocation isn't shared, it's just if we want a non-global config that is passed around the application it needs to happen at the parent level, at 'main'. Happy to discuss more.